var/spack/repos/builtin/packages/py-cupy/package.py

```diff
@@ -32,3 +32,12 @@ class PyCupy(PythonPackage):
     depends_on("nccl")
     depends_on("cudnn")
     depends_on("cutensor")
+
+    conflicts("~cuda")
+
+    def setup_build_envionment(self, env):
```

Comment:
```suggestion
    def setup_build_environment(self, env):
```

Reply:
:point_up: should hopefully fix your issue

Reply:
Thanks good catch.  After fixing this, I found one more bug.  should be ready to merge now.